src/test/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTimingTest.java

```diff
@@ -493,10 +507,210 @@ public void busyStepTest() throws Exception {
+        assertNotNull(stepStart);
+        GenericStatus status = StatusAndTiming.computeChunkStatus(b1, null, execution.getNode("2"), execution.getNode("5"), null);
+        assertEquals(GenericStatus.QUEUED, status);
+        ExecutorTaskInfoAction execAction = stepStart.getAction(ExecutorTaskInfoAction.class);
+        assertNotNull(execAction);
+        assertTrue(execAction.isQueued());
+        assertEquals("Waiting for next available executor", execAction.getWhyBlocked());
+
+        DumbSlave agent = j.createSlave("test-agent", "test", null);
+
+        SemaphoreStep.waitForStart("wait/1", b1);
+        status = StatusAndTiming.computeChunkStatus(b1, null, execution.getNode("2"), execution.getNode("8"), null);
+        assertEquals(GenericStatus.IN_PROGRESS, status);
+
+        stepStart = execution.getNode("5");
+        assertNotNull(stepStart);
+        execAction = stepStart.getAction(ExecutorTaskInfoAction.class);
+        assertNotNull(execAction);
+        assertTrue(execAction.isLaunched());
+        assertNull(execAction.getWhyBlocked());
+
+        SemaphoreStep.success("wait/1", null);
+        j.assertBuildStatusSuccess(j.waitForCompletion(b1));
+    }
+
+    @Issue("JENKINS-44981")
+    @Test
+    public void queuedAndCanceled() throws Exception {
+        WorkflowJob job = j.jenkins.createProject(WorkflowJob.class, "queuedAndCanceled");
+        job.setDefinition(new CpsFlowDefinition("stage('some-stage') {\n" +
+                "  node('test') {\n" +
+                "    echo 'hello'\n" +
+                "    semaphore 'wait'\n" +
+                "  }\n" +
+                "}\n", true));
+
+        WorkflowRun b1 = job.scheduleBuild2(0).waitForStart();
+        j.waitForMessage("Still waiting to schedule task", b1);
+
+        CpsFlowExecution execution = (CpsFlowExecution) b1.getExecutionPromise().get();
+
+        // node 2: FlowStartNode
+        // node 5: first StepStartNode for node
+        FlowNode stepStart = execution.getNode("5");
+        assertNotNull(stepStart);
+        GenericStatus status = StatusAndTiming.computeChunkStatus(b1, null, execution.getNode("2"), execution.getNode("5"), null);
+        assertEquals(GenericStatus.QUEUED, status);
+        ExecutorTaskInfoAction execAction = stepStart.getAction(ExecutorTaskInfoAction.class);
+        assertNotNull(execAction);
+        assertTrue(execAction.isQueued());
+        assertEquals("Waiting for next available executor", execAction.getWhyBlocked());
+
+        Queue.Item[] items = Queue.getInstance().getItems();
+        assertEquals(1, items.length);
+        assertEquals(job, items[0].task.getOwnerTask());
+        assertTrue(Queue.getInstance().cancel(items[0]));
+        j.assertBuildStatus(Result.FAILURE, j.waitForCompletion(b1));
+
+        status = StatusAndTiming.computeChunkStatus(b1, null, execution.getNode("2"), execution.getNode("5"), null);
+        assertEquals(GenericStatus.FAILURE, status);
```

Comment:
I do, yes, but that's not the existing behavior and I didn't want to break expectations there. It is definitely weird logic, though, so I'll revisit.